reporter/src/main/kotlin/model/EvaluatedModel.kt

```diff
@@ -19,7 +19,284 @@
+ *   result.
+ * * [RuleViolationResolution]s from the [ReporterInput.resolutionProvider] are matched against all [RuleViolation]s.
+ *
+ * The current implementation is missing these features:
+ *
+ * * [LicenseFindingCuration]s are not yet applied to the model.
+ *
+ * The model also contains useful containers to easily access some content of the [OrtResult], for example a list of
+ * all [OrtIssue]s in their [evaluated form][EvaluatedOrtIssue] which contains back-references to its source.
+ *
+ * The model can be serialized to JSON with the helper function [toJson]. It uses a special [JsonMapper] that
+ * de-duplicates objects in the result. For this it uses Jackson's [JsonIdentityInfo] to automatically generate [Int]
+ * IDs for the objects. All objects for which the model contains containers, like [issues] or [packages], are serialized
+ * only once in those containers. All other references to those objects are replaced by the [Int] IDs. This is required
+ * because the model contains cyclic dependencies between objects which would otherwise cause stack overflows during
+ * serialization, and it also reduces the size of the result file.
+ *
+ * Use cases for the [EvaluatedModel] are:
+ *
+ * * Input for the [WebAppReporter], so that it does not have to evaluate the model at runtime.
+ * * Input for [Reporter] implementations, so that they do not have to repeatedly implement the application of excludes,
+ *   resolutions, and so on.
+ * * Input for external tools, so that they do not have to re-implement the logic for evaluating the model.
+ *
+ * Important notes for working with this model:
+ *
+ * * The model uses Kotlin data classes with cyclic dependencies, therefore the [hashCode] and [toString] of affected
+ *   classes cannot be used, because they would create stack overflows.
+ * * When modifying the model make sure that the objects are serialized at the right place. By default Jackson
+ *   serializes an Object with [ObjectIdInfo] the first time the serializer sees the object. If this is not desired
+ *   because the object shall be serialized as the generated ID, the [JsonIdentityReference] annotation can be used to
+ *   enforce this. For example, the list of [EvaluatedOrtIssue]s is serialized before the list of [EvaluatedPackage]s.
+ *   Therefore [EvaluatedOrtIssue.pkg] is annotated with [JsonIdentityReference].
+ */
 data class EvaluatedModel(
-    val stats: Statistics
-    // TODO: add the actual evaluated data model.
+    val pathExcludes: List<PathExclude>,
+    val scopeExcludes: List<ScopeExclude>,
+    val copyrights: List<CopyrightStatement>,
+    val licenses: List<LicenseId>,
+    val scopes: List<ScopeName>,
+    val issueResolutions: List<IssueResolution>,
+    val issues: List<EvaluatedOrtIssue>,
+    val scanResults: List<EvaluatedScanResult>,
+    val packages: List<EvaluatedPackage>,
+    val paths: List<EvaluatedPackagePath>,
```

Comment:
Why is `paths` not removed in favor of a property of èvaluatedPackage`?

Reply:
This is a global collection of all paths for deduplication, because the same path can be referenced from different places.